hail/src/main/scala/is/hail/expr/ir/Compile.scala

```diff
@@ -133,3 +134,130 @@ object CompileWithAggregators {
     (ir.pType, f.asInstanceOf[(Int, Region) => (F with FunctionWithAggRegion)])
   }
 }
+
+object CompileIterator {
+
+  import is.hail.asm4s.joinpoint._
+
+  private trait StepFunctionBase {
+    def loadAddress(): Long
+  }
+
+  private trait TMPStepFunction extends AsmFunction3RegionLongIteratorRegionBoolean with StepFunctionBase
+
+  private abstract class LongIteratorWrapper extends Iterator[java.lang.Long] {
+    def step(): Boolean
+
+    protected val stepFunction: StepFunctionBase
+    private var _stepped = false
+    private var _hasNext = false
+
+    def hasNext: Boolean = {
+      if (!_stepped) {
+        _hasNext = step()
+        _stepped = true
+      }
+      _hasNext
+    }
+
+    def next(): java.lang.Long = {
+      if (!hasNext)
+        return Iterator.empty.next()
+      _stepped = false
+      stepFunction.loadAddress()
+    }
+  }
+
+  private def compileStepper[F >: Null <: StepFunctionBase : TypeInfo](
+    ctx: ExecuteContext,
+    body: IR,
+    argTypeInfo: Array[ParamType],
+    printWriter: Option[PrintWriter]
+  ): (PType, (Int, Region) => F) = {
+
+    val fb = EmitFunctionBuilder.apply[F](ctx, "stream", argTypeInfo.toFastIndexedSeq, CodeParamType(BooleanInfo))
+    val stepF = fb.apply_method
+    val stepFECB = stepF.ecb
+
+    val er = EmitRegion.default(stepF)
+    val emitter = new Emit(ctx, stepFECB)
+
+    var ir = body
+    ir = LoweringPipeline.compileLowerer(true)(ctx, ir).asInstanceOf[IR].noSharing
+    TypeCheck(ir)
+    InferPType(ir, Env.empty[PType])
+    val returnType = ir.pType.asInstanceOf[PStream].elementType.asInstanceOf[PStruct].setRequired(true)
+
+    val optStream = EmitStream.emit(emitter, ir, stepF, er.region, Env.empty, None);
+
+    val elementAddress = stepF.genFieldThisRef[Long]("elementAddr")
+
+    val didSetup = stepF.genFieldThisRef[Boolean]("didSetup")
```

Comment:
This is never initialized. It's probably better not to rely on Java's default initialization behavior. Add `didSetup := false` to the class initializer?